### ISSUE TITLE
Support using client key and cert for OAuth1 with ssl

### ISSUE THREAD

**Issue by george-haddad** [omouse/node-oauth-libre#17]
There are some scenarios (interfacing with xero accounting apis) where we need to specify a client ssl key and cert in the https connection. Usually would do this by adding it to the options object in  _createClient() function. The original ciaranj/node-oauth code doesn't have it but it's been requested and a pull-request made (but not merged)

Can this pull request be added here? It would be great.
https://github.com/ciaranj/node-oauth/pull/302

**Comment by omouse** [comment]
@george-haddad I'll clean up the PR a bit before merging, **yes we can do it.**

Do you have an example of how you're using HTTP options? When looking at the node docs it looks like the key and cert options aren't documented? https://nodejs.org/api/net.html#net_socket_connect_options_connectlistener

**Comment by george-haddad** [comment]
@omouse thanks for looking into this. Yes so below is an example of how I would use it.
```
const options = {
    cert: fs.readFileSync('./lib/utils/keys/my_publickey.cer', 'utf8'),
    key: fs.readFileSync('./lib/utils/keys/my_pem.key', 'utf8')
};

oauth.setHttpOptions(options);
```
There is documentation about this for the nodejs https class.
[https://nodejs.org/api/https.html#https_https_globalagent](https://nodejs.org/api/https.html#https_https_globalagent)

**Comment by omouse** [comment]
@george-haddad awesome! I just read through and noticed that the httpModel changes so that the httpOptions that are accepted will change. I'll add a comment in your PR to explain that and otherwise it's good to go.

We can also update the README with your code snippet to show others how to use it.

**Comment by caco0516** [comment]
@omouse , Is this issue closed with pull request #18 ?

**Comment by omouse** [comment]
yep it's good to go, closing the issue :+1:
